#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From: Albert Astals Cid <aacid@kde.org>
Date: Wed, 26 Mar 2025 11:26:32 +0100
Subject: [PATCH 3/7] Make sure regex doesn't stack overflow by limiting it

Happens with very long pdfsubver strings when compiled with
-fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -flto=auto

(cherry picked from commit f54b815672117c250420787c8c006de98e8c7408)

CVE-2025-43718
---
 poppler/PDFDoc.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/poppler/PDFDoc.cc b/poppler/PDFDoc.cc
index 814c3095..894b6767 100644
--- a/poppler/PDFDoc.cc
+++ b/poppler/PDFDoc.cc
@@ -490,7 +490,7 @@ static PDFSubtypePart pdfPartFromString(PDFSubtype subtype, const std::string &p

 static PDFSubtypeConformance pdfConformanceFromString(const std::string &pdfsubver)
 {
-    const std::regex regex("PDF/(?:A|X|VT|E|UA)-[[:digit:]]([[:alpha:]]+)");
+    const std::regex regex("PDF/(?:A|X|VT|E|UA)-[[:digit:]]([[:alpha:]]{1,3})");
     std::smatch match;
     PDFSubtypeConformance pdfConf = subtypeConfNone;

--
2.47.2
